modules/smtp/strong_password/files/smtp_honeypot.py

```diff
@@ -0,0 +1,346 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Credits : https://github.com/awhitehatter/mailoney
+"""
+
+import socket
+import sys
+import errno
+import time
+import threading
+import datetime
+import asyncore
+import asynchat
+import os
+import json
+
+
+MAILSERVER_NAME = "OWASP Python Honeypot"
+BIND_IP='0.0.0.0'
+BIND_PORT=587
+LOGFILE = '/root/logs/ohp_smtp_honeypot_logs.txt'
+__version__ = 'ESMTP Exim 4.69 #1 Thu, 29 Jul 2010 05:13:48 -0700'
+EMPTYSTRING = ''
+NEWLINE = '\n'
+
+output_lock = threading.Lock()
+hpc,hpfeeds_prefix = None,""
+
+def log_to_file(ip, port, data):
+    output_lock.acquire()
+    try:
+        logfile_handle = open(LOGFILE, "a")
+        logfile_handle.write(
+            json.dumps(
+                {
+                    "ip": ip,
+                    "data": data,
+                    "port": port,
+                    "module_name": "smtp/mail_honeypot",
+                    'date': datetime.datetime.now().\
+                    strftime("%Y-%m-%d %H:%M:%S")
+                }
+            ) + "\n"
+        )
+        logfile_handle.close()
+    finally:
+        output_lock.release()
+    #print(file_path,ip,port,data)
+    #with output_lock:
+        #with open(file_path, "a") as f:
+    message = "[{0}][{1}:{2}] {3}".format(time.time(), ip, port, data)
+    #print(message)
+
+
+
+class SMTPChannel(asynchat.async_chat):
+    COMMAND = 0
+    DATA =1
+
+    def __init__(self, server, conn, addr):
+        asynchat.async_chat.__init__(self, conn)
+        self.__rolling_buffer = ""
+        self.__server = server
+        self.__conn = conn
+        self.__addr = addr
+        self.__line = []
+        self.__state = self.COMMAND
+        self.__greeting = 0
+        self.__mailfrom = None
+        self.__rcpttos = []
+        self.__data = ''
+        self.__fqdn = MAILSERVER_NAME
+        try:
+            self.__peer = conn.getpeername()
+        except socket.error as err:
+            # a race condition  may occur if the other end is closing
+            # before we can get the peername
+            self.close()
+            if err[0] != errno.ENOTCONN:
```

Comment:
`err` here is not an array.

Reply:
@ChakshuGupta : according to the examples here socket.error gives a list of Error no and Message.

Reply:
According to this https://docs.python.org/3/library/socket.html#socket.error it is an alias for OSError https://docs.python.org/3/library/exceptions.html#OSError which is an object not a list. Also, it is deprecated now.